### PR TITLE
SQLite open database recipe

### LINKED ISSUE #453: Recipe: Creating Database in File System
|  |  |
| ------------- | ------------- |
| Crates  | rusqlite|
| Section  | 7.1 SQLite  |

This recipe should demonstrate using rusqlite by saving the database to the file system.

### PR DESCRIPTION
**Pull request by lnicola**
fixes #453, #454

Based on #460. I'll rebase afterwards, please ignore the first two commits.

CC @robertDurst as per #455.

--------------------------
### Things to check before submitting a PR

- [x] the tests are passing locally with `cargo test`
- [x] cookbook renders correctly in `mdbook serve -o`
- [x] commits are squashed into one and rebased to latest master
- [x] PR contains correct "fixes #ISSUE_ID" clause to autoclose the issue on PR merge
- [x] spell check runs without errors `./ci/spellchecker.sh`
- [x] link check runs without errors `link-checker ./book`
- [x] non rendered items are in sorted order (links, reference, identifiers, Cargo.toml)
- [x] links to docs.rs have wildcard version `https://docs.rs/tar/*/tar/struct.Entry.html`
- [x] example has standard [error handling](https://rust-lang-nursery.github.io/rust-cookbook/about.html#a-note-about-error-handling)
- [x] code identifiers in description are in hyperlinked backticks

### Things to do after submitting PR
- [x] check if CI is happy with your PR

### CODE REVIEW

**Comment by lnicola** [comment]
Should I drop `error-chain` from this example, like in #433?

**Comment by lnicola** [comment]
Right. Now what?

- use `bundled` everywhere
- use `bundled` on Windows
- figure out how to include SQLite in the AppVeyor build
- ignore the test

The third option is probably the best, but I don't know anything about the AppVeyor configuration.

**Comment by lnicola** [comment]
Possibly related to https://github.com/budziq/rust-skeptic/issues/87.

Can we just `ignore` these tests?

File: src/database/sqlite/initialization.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+## Create a SQLite database
+
+[![rusqlite-badge]][rusqlite] [![cat-database-badge]][cat-database]
+
+You can use the `rusqlite` crate to open SQLite databases. Note that, if you
```
Comment (AndyGauge) [inline src/database/sqlite/initialization.md]: Personal pronouns should not be used in the description.  Refer to the code not the author or the audience.  In this case, just drop 'You can'

File: src/database/sqlite/initialization.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+are compiling on Windows, you'll have to perform some [additional steps][linking]
```
Comment (AndyGauge) [inline src/database/sqlite/initialization.md]: See the [bundled] feature for compiling in Windows.

File: src/database/sqlite/initialization.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        &[],
+    )?;
+
+    Ok(())
+}
+```
+
+[`Connection::open`]: https://docs.rs/rusqlite/*/rusqlite/struct.Connection.html#method.open
+
+[linking]: https://github.com/jgallagher/rusqlite#user-content-notes-on-building-rusqlite-and-libsqlite3-sys
```
Comment (AndyGauge) [inline src/database/sqlite/initialization.md]: This needs to match the link above (documentation)

File: src/links.md [hunk 4]
```diff
@@ -114,7 +120,7 @@ Keep lines sorted.
 [serde-json-badge]: https://badge-cache.kominick.com/crates/v/serde_json.svg?label=serde_json
 [serde-json]: https://docs.rs/serde_json/*/serde_json/
 [serde]: https://docs.rs/serde/
-[std-badge]: https://badge-cache.kominick.com/badge/std-1.25.0-blue.svg
+[std-badge]: https://badge-cache.kominick.com/badge/std-1.29.1-blue.svg
```
Comment (AndyGauge) [inline src/links.md]: 👍

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -20,7 +20,7 @@ cd rust-cookbook
 Cookbook is built with [mdBook], so install that first with Cargo:
```
Comment (AndyGauge) [inline CONTRIBUTING.md]: git checkout origin/master -- CONTRIBUTING.md

File: src/science.md [hunk 6]
```diff
@@ -5,9 +5,18 @@
 | [Creating complex numbers][ex-creating-complex-numbers] | [![num-badge]][num] | [![cat-science-badge]][cat-science] |
 | [Adding complex numbers][ex-adding-complex-numbers] | [![num-badge]][num] | [![cat-science-badge]][cat-science] |
 | [Mathematical functions on complex numbers][ex-mathematical-functions] | [![num-badge]][num] | [![cat-science-badge]][cat-science] |
+| [Calculating the side length of a triangle][ex-calculating-side-length-of-triangle] | [![std-badge]][std] | [![cat-science-badge]][cat-science] |
```
Comment (AndyGauge) [inline src/science.md]: can you git checkout origin/master -- src/science.md and the other science files and push this?  I need clean commit history

File: src/database/sqlite/initialization.md (hunk 3 above)
Comment (lnicola) [reply src/database/sqlite/initialization.md]: Fixed

**Comment by lnicola** [comment]
I rebased again. I don't know why those commits showed up -- I don't thing they were there before. Can you check?

**Pull request merged**